Content.Shared/Chemistry/Reagent/ReagentId.cs

```diff
@@ -20,17 +21,18 @@ public partial struct ReagentId : IEquatable<ReagentId>
     /// Any additional data that is unique to this reagent type. E.g., for blood this could be DNA data.
     /// </summary>
     [DataField("data")]
-    public ReagentData? Data { get; private set; }
+    public List<ReagentData> Data { get; private set; } = new();
```

Comment:
Wouldn't it make more sense to store reagent data as comps on an entity and make this field a nullable entity ID instead?

Reply:
I've tried to base the implementation on the structure used in the original #19323 PR (though expanded to a list to allow multiple ReagentDatas). Since the PR was merged I was under the assumption that this was seen as the desirable implementation.